### PR TITLE
Feature/resource management improvements

### PR DESCRIPTION
**Pull request by yalania**
Added some menu actions for the filesystem and fixing a bug with the normal texures:

- Rename
- Delete
- Extract Prefab
- Enable importing thread

### CODE REVIEW

File: Engine/EditorUI/Panel/PanelProjectExplorer.cpp [hunk 1]
```diff
@@ -117,6 +119,7 @@ void PanelProjectExplorer::ShowFoldersHierarchy(const Path& path)
 				filename = ICON_FA_FOLDER_OPEN " " + path_child->GetFilename();
 			}
 			bool expanded = ImGui::TreeNodeEx(filename.c_str(), flags);
+			ResourceDropTarget(path_child);
 			if (expanded) {
```
Comment (marcorod94) [inline Engine/EditorUI/Panel/PanelProjectExplorer.cpp]: code Style

File: Engine/EditorUI/Panel/PanelProjectExplorer.h [hunk 2]
```diff
@@ -26,17 +26,21 @@ class PanelProjectExplorer : public Panel
 	void FilesDrop() const;
 	void ResourceDragSource(Metafile* file) const;
 
+	void ResourceDropTarget(Path * folder_path) const;
```
Comment (marcorod94) [inline Engine/EditorUI/Panel/PanelProjectExplorer.h]: Code Style

File: ImportedResourcesCleaner.ps1 [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Get-ChildItem . -recurse -include *.meta | remove-item
```
Comment (mariofv) [inline ImportedResourcesCleaner.ps1:1]: I think that is better if we don't have this first line atm, as we dont want to delete by accident metafiles in the Game project. We can add this after the split of Assets is done in a new repository.

**Review by mariofv — APPROVED** [review]

File: Engine/Module/ModuleFileSystem.h [hunk 4]
```diff
@@ -33,13 +32,19 @@ class ModuleFileSystem : public Module
 	bool Remove(const std::string& path);
 
 	Path* Copy(const std::string& source_path, const std::string& destination_path, const std::string& copied_file_name = std::string());
+	Path * Move(const std::string & source_path, const std::string & destination_path);
```
Comment (marcorod94) [inline Engine/Module/ModuleFileSystem.h]: Code style

File: Engine/Module/ModuleResourceManager.cpp [hunk 5]
```diff
@@ -94,40 +94,80 @@ bool ModuleResourceManager::CleanUp()
...
-		thread_comunication.thread_importing_hash = std::hash<std::string>{}(path_child->GetFullPath());
-		while (thread_comunication.main_importing_hash == std::hash<std::string>{}(directory_path.GetFullPath()))
+		if (path_child->IsDirectory())
 		{
-			Sleep(1000);
+			CleanMetafilesInDirectory(*path_child);
+		}
+		else if (path_child->IsMeta())
+		{
+			if (!metafile_manager->IsMetafileConsistent(*path_child))
```
Comment (marcorod94) [inline Engine/Module/ModuleResourceManager.cpp]: We could put this in one it instead of two

File: Engine/Module/ModuleResourceManager.h [hunk 6]
```diff
@@ -167,7 +168,7 @@ class ModuleResourceManager : public Module
 	std::unique_ptr<ResourceDataBase> resource_DB = nullptr;
 
 private:
-	const size_t importer_interval_millis = 60*10*1000;
+	const size_t importer_interval_millis = 15*1000;
```
Comment (marcorod94) [inline Engine/Module/ModuleResourceManager.h]: code style

File: Engine/ResourceManagement/Metafile/MetafileManager.cpp [hunk 7]
```diff
@@ -72,26 +72,31 @@ Metafile* MetafileManager::CreateMetafile(Path& asset_file_path, ResourceType re
...
 	created_metafile->version = Importer::IMPORTER_VERSION;
 
+	SaveMetafile(created_metafile, asset_file_path);
+
+	metafiles[created_metafile->metafile_path] = created_metafile;
+
+	return created_metafile;
+}
+
+void MetafileManager::SaveMetafile(Metafile * created_metafile, Path & asset_file_path) const
```
Comment (marcorod94) [inline Engine/ResourceManagement/Metafile/MetafileManager.cpp]: Code Style

File: Engine/ResourceManagement/Metafile/MetafileManager.h [hunk 8]
```diff
@@ -16,11 +16,13 @@ class MetafileManager
 
 	Metafile* CreateMetafile(const std::string& asset_file_path_string, ResourceType resource_type);
 	Metafile* CreateMetafile(Path& asset_file_path, ResourceType resource_type, uint32_t uuid = 0);
+
+	void SaveMetafile(Metafile * created_metafile, Path & asset_file_path) const;
```
Comment (marcorod94) [inline Engine/ResourceManagement/Metafile/MetafileManager.h]: Code Style

**Review by marcorod94 — APPROVED** [review]
Really nice job!!

**Pull request merged**